pages/k8s/tigera-secure-ee.md

```diff
@@ -0,0 +1,212 @@
+---
+wrapper_template: "base_docs.html"
+markdown_includes:
+  nav: "shared/_side-navigation.md"
+context:
+  title: "Using Tigera Secure EE"
+  description: Using Tigera Secure EE with CDK
+keywords: network, cni, tigera
+tags: [operating]
+sidebar: k8smain-sidebar
+permalink: tigera-secure-ee.html
+layout: [base, ubuntu-com]
+toc: False
+---
+
+Tigera Secure EE is a software-defined network solution that can be used with
+Kubernetes. For those familiar with Calico, Tigera Secure EE is essentially
+Calico with enterprise features on top.
+
+Support for Tigera Secure EE in CDK is provided in the form of a
+tigera-secure-ee subordinate charm, which can be used instead of flannel or
+calico.
+
+## Deploying CDK with Tigera Secure EE
+
+Before you start, you will need:
+* Tigera Secure EE license key
+* Tigera private Docker registry credentials (provided as a Docker config.json)
+
+NOTE: Tigera Secure EE's network traffic, much like Calico's, is filtered on
+many clouds. It should work on MaaS, and can work on AWS if you manually
+configure instances to disable source/destination checking.
+
+To start, deploy CDK with Tigera Secure EE:
+
+```bash
+juju deploy cs:~containers/kubernetes-tigera-secure-ee
+```
+
+Configure the tigera-secure-ee charm with your license key and registry
+credentials:
+
+```bash
+juju config tigera-secure-ee \
+  license-key=$(base64 -w0 license.yaml) \
+  registry-credentials=$(base64 -w0 config.json)
+```
+
+Wait for the deployment to settle before continuing on.
+
+## Using the built-in elasticsearch-operator
+
+NOTE: elasticsearch-operator is only recommended for testing or demonstrative
+purposes. For production deployments, please skip down to the next section.
+
+For testing and quick start purposes, the tigera-secure-ee charm deploys
+[elasticsearch-operator] into your Kubernetes cluster by default. For it to
+properly work, you will need to create a StorageClass.
+
+The easiest way to do this is with the hostpath provisioner. Create a file named
+`elasticsearch-storage.yaml` containing the following:
+
+```yaml
+# This manifest implements elasticsearch-storage using local host-path volumes.
+# It is not suitable for production use; and only works on single node clusters.
+# For production or multinode clusters, use a different elasticsearch-storage implementation.
```

Comment:
"use a different elasticsearch-storage implementation"

My first thought was, "how?" Is it just telling me to use real storage instead of hostpath?

Reply:
Good point, thanks. I removed this line - I think it was a bad place to mention alternatives anyway. The alternatives seem adequately covered elsewhere in the doc. Please let me know if anything still seems off.

Reply:
Looks good, thanks.